src/pages/auth/join.tsx

```diff
@@ -1,14 +1,63 @@
+import AcceptTerms from 'components/Auth/AcceptTerms';
 import SignupForm from 'components/Auth/SignupForm';
 import ThemeSelector from 'components/Auth/ThemeSelector';
+import { useState } from 'react';
 import styled from 'styled-components';
 import { theme } from 'styles/theme';
+import { UserSignupRequest } from 'types/user';
 
 function Join() {
+  const [signupInfo, setSignupInfo] = useState<UserSignupRequest>({
+    name: { value: null, isComplete: false },
+    email: { value: null, isComplete: false },
+    emailConfirm: { value: null, isComplete: false },
+    password: { value: null, isComplete: false },
+    passwordConfirm: { value: null, isComplete: false },
+    nickname: { value: null, isComplete: false },
+    themePreference: { value: [], isComplete: false },
+  });
+
+  const handleComplete = (type: keyof UserSignupRequest, value: boolean) => {
+    setSignupInfo({ ...signupInfo, [type]: { ...signupInfo[type], isComplete: value } });
+  };
```

Comment:
```suggestion
  const handleComplete = useCallback((type: keyof UserSignupRequest, value: boolean) => {
    setSignupInfo((prevSignupInfo) => ({
      ...prevSignupInfo,
      [type]: { ...prevSignupInfo[type], isComplete: value },
    }));
  }, []);
```

handleComplete 함수가 하위 컴포넌트에서 props로 넘겨지면서 사용되고 있는데 join 컴포넌트가 리렌더링 될 때마다 handleComplete도 같이 새로 생성된단말이죠? 

하지만 handleComplete는 리렌더링되더라도 함수가 새로 만들어질 필요가 없어요.
type과 value를 받아서 기존 signUpInfo 값만 갱신시켜주면 되니까요.

이 때 useCallback을 사용할 수 있어요.
useCallback은 두번째 인자로 의존성 배열을 받는데 이 경우에는 함수 재생성에 영향을 받는 값이 없기 때문에 빈 배열을 전달시켜주면 될 것 같네요!!

이렇게 하게 되면 하위컴포넌트에서 handleComplete를 props로 받아서 사용하는 경우에 useEffect 의존성 배열에 handleComplete를 넣더라도 저번에 말씀하신 것처럼 무한 렌더링되지 않을 수 있어요!

밑에 있는 다른 handle~~ 함수들도 동일한 맥락같아요!